runtime/src/accounts_db.rs

```diff
@@ -1091,6 +1118,20 @@ impl AccountsDB {
         slots.sort();
         let mut accounts_index = self.accounts_index.write().unwrap();
         for slot_id in slots.iter() {
+            let storage_maps: Vec<Arc<AccountStorageEntry>> = self
+                .storage
+                .read()
+                .unwrap()
+                .0
+                .get(&slot_id)
+                .unwrap_or(&HashMap::new())
+                .values()
+                .cloned()
+                .collect();
+            for storage in storage_maps.into_iter() {
+                storage.restore_account_count();
```

Comment:
Note [this comment](https://github.com/solana-labs/solana/pull/7013/files#r348980406).

Reply:
This code is tightly paired with this [code](https://github.com/solana-labs/solana/pull/7013/files#diff-2099c5256db4eb5975c8834af38f6456R1182).